components/EntryForm.js

```diff
@@ -50,23 +22,59 @@ export default function EntryForm() {
     { label: "Diesel", value: "diesel" },
     { label: "Hybrid", value: "hybrid" },
     { label: "Electric-Strommix", value: "electric-strommix" },
-    { label: "Electric-Renewable", value: "electric-renewable" }
+    { label: "Electric-Renewable", value: "electric-renewable" },
   ];
 
+  function handleDropdownChange(event) {
+    setTransport(event.target.value);
+  }
+
+  function handleDropdownChangeFuel(event) {
+    setFuel(event.target.value);
+  }
+
+  function handleKm(event) {
+    setKm(event.target.value);
+  }
+
+  function handleCalculateCo2(transport, km, fuel) {
+    const selectedTransport = transport === "car" ? fuel : transport;
+    return calculator[selectedTransport](km);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const newEntry = Object.fromEntries(formData);
+    const transport = newEntry.transport;
+    const km = newEntry.km;
+    const fuel = newEntry.fuel;
+    const result = handleCalculateCo2(transport, km, fuel);
+    newEntry.result = result;
+    console.log(newEntry);
+    console.log(result);
```

Comment:
Remove `console.logs` when you're done inspecting the behavior of your code.

Reply:
I have removed them.